includes/pages/admin_import.php

```diff
@@ -325,49 +337,83 @@ function prepare_events($file, $shifttype_id, $add_minutes_start, $add_minutes_e
       if ($shift['shifttype_id'] != $tmp['shifttype_id'] || $shift['title'] != $tmp['title'] || $shift['start'] != $tmp['start'] || $shift['end'] != $tmp['end'] || $shift['RID'] != $tmp['RID'] || $shift['URL'] != $tmp['URL'])
         $shifts_updated[] = $shift;
     }
-  
+
   $shifts_deleted = array();
   foreach ($shifts_db as $shift)
     if (! isset($shifts_pb[$shift['PSID']]))
       $shifts_deleted[] = $shift;
-  
+
   return array(
       $shifts_new,
       $shifts_updated,
-      $shifts_deleted 
+      $shifts_deleted
   );
 }
-
 function read_xml($file) {
   global $xml_import;
   if (! isset($xml_import))
     $xml_import = simplexml_load_file($file);
   return $xml_import;
 }
-
 function shifts_printable($shifts, $shifttypes) {
   global $rooms_import;
   $rooms = array_flip($rooms_import);
-  
+
   uasort($shifts, 'shift_sort');
-  
+
   $shifts_printable = array();
   foreach ($shifts as $shift)
     $shifts_printable[] = array(
         'day' => date("l, Y-m-d", $shift['start']),
         'start' => date("H:i", $shift['start']),
         'shifttype' => ShiftType_name_render([
             'id' => $shift['shifttype_id'],
-            'name' => $shifttypes[$shift['shifttype_id']] 
+            'name' => $shifttypes[$shift['shifttype_id']]
         ]),
         'title' => shorten($shift['title']),
         'end' => date("H:i", $shift['end']),
-        'room' => $rooms[$shift['RID']] 
+        'room' => $rooms[$shift['RID']]
     );
   return $shifts_printable;
 }
 
 function shift_sort($a, $b) {
   return ($a['start'] < $b['start']) ? - 1 : 1;
 }
+
+function export_xls() {
+	$filename = $fn = tempnam ('/tmp', '.csv'); //  Temporary File Name
```

Comment:
Whitespace `tempnam ('/tmp', '.csv')`.
Is the variable `$fn` used anywhere? If not, delete it.